report/4.Design/Dependability.tex

```diff
@@ -1,25 +1,51 @@
 \section{Dependability}\label{Design:Dependability}
 
 \subsection{Fault Tolerance - Bit-flipping}
-One issue that may Implement redundancy directly in application
+As a proof of concept, the group decided to implement bit-flip fault tolerance in the laser module on the NXT, given the following rules.
 
-\begin{description}[style=nextline]
-  \item[Every variable \texttt{x} must be duplicated \texttt{x0}] \cite{errorDetectionSoft}
-  \item[Every write operation perfomed on \texttt{x}, must also be performed on \texttt{x0}] bang \cite{errorDetectionSoft}
-  \item[After each read of \texttt{x}, consistency must be checked between \texttt{x} and \texttt{x0}] bang \cite{errorDetectionSoft}
-  \item[For every conditional, repeat test in every target (both then and else branches)] bang \cite{errorDetectionSoft}
-\end{description}
+\begin{enumerate}
+  \item Every variable \texttt{x} must be duplicated \texttt{x0} \cite{errorDetectionSoft}
+  \item Every write operation perfomed on \texttt{x}, must also be performed on \texttt{x0} \cite{errorDetectionSoft}
+  \item After each read of \texttt{x}, consistency must be checked between \texttt{x} and \texttt{x0} \cite{errorDetectionSoft}
+  \item For every conditional, repeat test in every target (both then and else branches) \cite{errorDetectionSoft}
+\end{enumerate}
+
+Following the four rules above, the code could look like below, taking care of reading from variables, if-conditions and variable duplication.
 
 \noindent\begin{minipage}{.45\textwidth}
-\begin{lstlisting}[caption={code 1\cite{errorDetectionSoft}},frame=tlrb,numbers=none]{Without Bitflipping A}
-a = b;
+\begin{lstlisting}[caption={Without bitflip-security},frame=tlrb,numbers=none]
```

Comment:
Here we have actual implementation details in the design chapter. Is this not wrong?

Reply:
Måske.. Det er et KORT uddrag, for at vise konceptet..